Library/Formula/aws-sdk-cpp.rb

```diff
@@ -0,0 +1,70 @@
+class AwsSdkCpp < Formula
+  desc "AWS SDK for C++"
+  homepage "https://github.com/aws/aws-sdk-cpp"
+  url "https://github.com/aws/aws-sdk-cpp/archive/0.9.6.tar.gz"
+  sha256 "cf2dad62e50c5c5cff88a4ad959af3aa402aa7e8aa1692bdf807052c0fe4aeb9"
+  head "https://github.com/aws/aws-sdk-cpp.git"
+
+  ALL_CLIENTS = [
```

Comment:
Oh, I didn't realize the CMake flag was `BUILD_ONLY`.  So you're right that due to this CMake configuration, it currently behaves as you want it to.  But this CMake config sort of reverses how options work in Homebrew, and I think it will be confusing for users.  For the vast majority of formula, adding an option to build X doesn't _remove_ Y, like it can here.